src/test/java/harness/CRLJanitor.java

```diff
@@ -51,4 +81,80 @@ private static GoogleCredentials getSACredentials() {
       throw new RuntimeException("Error reading SA credentials for Janitor client.", ioEx);
     }
   }
+
+  private static Publisher initializeJanitorPubsubPublisher() {
+    TopicName topicName =
+        TopicName.of(
+            TestConfig.get().getJanitorPubSubProjectId(), TestConfig.get().getJanitorPubSubTopic());
+    try {
+      return Publisher.newBuilder(topicName)
+          .setCredentialsProvider(FixedCredentialsProvider.create(getSACredentials()))
+          .build();
+    } catch (IOException e) {
+      throw new JanitorException("Failed to initialize Janitor pubsub publisher.", e);
+    }
+  }
+
+  /**
+   * If Janitor is enabled, register the given workspace to be cleaned up. Otherwise return
+   * immediately.
+   */
+  public static void registerWorkspaceForCleanupIfEnabled(
+      UFWorkspace workspace, TestUser testUser) {
+    if (!TestConfig.get().getUseJanitorForExternalResourcesCreatedByTests()) {
+      return;
+    }
+    registerWorkspaceForCleanup(workspace, testUser);
+  }
+
+  /**
+   * Register a given workspace to be cleaned up. Wrapper around {@link
+   * #registerWorkspaceForCleanup(UUID, String, String)}
+   */
+  public static void registerWorkspaceForCleanup(UFWorkspace workspace, TestUser testUser) {
```

Comment:
You have 3 methods:

1. registerWorkspaceForCleanupIfEnabled
2. registerWorkspaceForCleanup
3. registerWorkspaceForCleanup takes in instanceId

Callers should only call 1, right? Can these three combined into method?

```
  public static void registerWorkspaceForCleanup(
      UFWorkspace workspace, TestUser testUser) {
    if (!TestConfig.get().getUseJanitorForExternalResourcesCreatedByTests()) {
      return;
    }
    ...
```

3 throws an exception if `UseJanitorForExternalResourcesCreatedByTests` isn't true. I don't think we should throw an exception if `UseJanitorForExternalResourcesCreatedByTests` isn't true. We should be able to turn off `UseJanitorForExternalResourcesCreatedByTests` indepedent of code.

Reply:
Good point. I wrote all three thinking they would be useful, but after changing the tests I think this only needs the one method.